### PR TITLE
Update window code

### LINKED ISSUE #3515: (WEB REPORT BY: pjb3005 REMOTE: 198.245.63.50:7777) Can't pull windows in the direction they're facing
>Revision a3d8d26835ddf904da2046864f5d9f71d1e939d6 

>General description of the issue
Title

>What you expected to happen
the window would be pulled just fine

>What actually happened
it didn't get pulled

>Steps to reproduce if possible
1. make a loose (1 dir)window
2. attempt to pull it into the direction it's facing

### PR DESCRIPTION
**Pull request by ghost**
That code was old and fucky, goddamn
- Span, formatting, newline, visible_message, yeah
- Separate sprites for window panes and full windows. Might need a bit more refining still, but that's that
- Remove deprecated commented silicate code. If anyone ever wants to include something akin to silicate, they would benefit from just recoding it from scratch. An item that adds health and potentially changes sprite color can't be that hard to make
- Reduce normal window health by 4 to 10. Why would you use plain windows
- Remove stupid paths that caused shitty inheritance. window/basic and window/full/basic were dumb as fucking shit and no longer exist (both transferred to their respective tree parent, for whatever variables needed transplanting, so 2-3). All full window types under full/ all reinforced window types under reinforced/, all in that order
- Add in more examine() information (construction state and health)
- Add in healthcheck() to replace destroy(), hit() and all that logging and variable juggling bullshit in about every proc where the window can be damaged
- ex_act(), blob_act(), meteorhit() now deplete health only and no longer magically delete or Destroy() windows. Looks a bit ugly on ex_act 3 since reinforced windows can't be destroyed but grilles always are, and plasma windows still don't block explosions even if they're now almost bomb-proof, but that's progress
- Remove window damage on passive grab window "slam", massively increase mob damage on neck/kill window slam (if you let yourself get grabbed that far, you deserve to die)
- Change construction from that ugly variable juggling to a fancy and functional switch() construction code. Include bitflags and deconstruction safety
- All files that were modified outside of window.dm and fullwindow.dm are compatibility fixes

Try as I might, I couldn't address #3515. It appears to be fundamentally fucked due to the way pushing objects (you enter the window's tile) and pulling objects (you cannot reach the other side of the window, ever) works. Maybe someone else will manage it

### CODE REVIEW

**Comment by dylanstrategie** [comment]
I love the smell of fresh merge conflicts in the morning

**Comment by ComicIronic** [comment]
>Reduce normal window health by 4 to 10. Why would you use plain windows

You really need to stop making balance changes in system PRs. These will block merges if they need to be discussed, but mostly they're not relevant to the system code you're changing, and they don't need to be included in the PR because they're not critical.

File: code/WorkInProgress/Cael_Aislinn/sculpture.dm [hunk 1]
```diff
@@ -144,7 +144,7 @@
 		spawn()
 			while(get_turf(src) != target_turf && num_turfs > 0)
 				for(var/obj/structure/window/W in next_turf)
-					W.destroy()
+					W.Destroy()
```
Comment (ComicIronic) [inline code/WorkInProgress/Cael_Aislinn/sculpture.dm]: Is this sufficient? As far as I know, Destroy doesn't call del().

Comment (dylanstrategie) [inline code/WorkInProgress/Cael_Aislinn/sculpture.dm]: Destroy() should be working as intended. There's a ..() call, I'm simply adding shit dropping on destruction here

Comment (ComicIronic) [inline code/WorkInProgress/Cael_Aislinn/sculpture.dm]: And I don't think Destroy calls del, since del calls Destroy. Have you tested this?

**Comment by dylanstrategie** [comment]
Shit, alright. I'll rebase it, one second

**Comment by clusterfack** [comment]
All them merges

**Comment by PJB3005** [comment]
I don'tcare about the lower normal window damage, it really doesn't matter

**Comment by clusterfack** [comment]
@PJB3005 he also made windows take more hits for blobs to kill

**Comment by PJB3005** [comment]
oh, well, uh, I've never played blob so I can't really say anything to that

**Comment by d3athrow** [comment]
>Reduce normal window health by 4 to 10. Why would you use plain windows

Dont really like that as it was buffed not that long ago
Also you could add repairing windows, one of the decent things paprika added to tg.

**Comment by dylanstrategie** [comment]
Should I change it to 15 then ?

Also repairing windows should most likely be done with silicate, maybe in chemical form again, or in item form (silicate paste or something), but it's mostly out of the scope of this PR

**Comment by PJB3005** [comment]
repairing windows makes more sense like this: welder on secured window repairs

**Comment by dylanstrategie** [comment]
I'm a sucker for silicate, but window repairing isn't urgent. Also welder to secured window is already used by many, MANY people to bash through windows

**Comment by PJB3005** [comment]
make it harm intent based

**Comment by dylanstrategie** [comment]
I think it would be better to use something like a silicate paste to use to repair and reinforce windows, but it really isn't urgent. Very few people would bother to repair windows, especially since the damage isn't visible unless you bother to examine panes

**Comment by dylanstrategie** [comment]
Needs some form of decision I guess

**Pull request merged**